File: app-miyou/src/main/java/com/miyou/app/application/monitoring/port/RagQualityMetricsPort.java
package com.miyou.app.application.monitoring.port;

public interface RagQualityMetricsPort {
	void recordMemoryCandidateCount(int count);

	void recordMemoryFilteredCount(int count);

	void recordMemorySimilarityScore(double score);

	void recordMemoryImportanceScore(double importance);

	void recordDocumentRelevanceScore(double score);
}

File: app-miyou/src/main/kotlin/com/miyou/app/application/monitoring/port/RagQualityMetricsPort.kt
package com.miyou.app.application.monitoring.port

interface RagQualityMetricsPort {
    fun recordMemoryCandidateCount(count: Int)

    fun recordMemoryFilteredCount(count: Int)

    fun recordMemorySimilarityScore(score: Double)

    fun recordMemoryImportanceScore(importance: Double)

    fun recordDocumentRelevanceScore(score: Double)
}
